java/core/src/main/java/com/microsoft/bond/ListBondType.java

```diff
@@ -98,13 +107,14 @@ protected final void serializeValue(SerializationContext context, List<TElement>
         this.verifyNonNullableValueIsNotSetToNull(value);
         int count = value.size();
         context.writer.writeContainerBegin(count, this.elementType.getBondDataType());
-        for (int i = 0; i < count; ++i) {
+        int i = 0;
+        for (TElement element : value) {
```

Comment:
That is: I'd like to avoid an overhead of a separate PR.

---
In reply to: [128562299](https://github.com/Microsoft/bond/pull/532#discussion_r128562299) [](ancestors = 128562299,128378679)